fire/docstrings.py

```diff
@@ -177,6 +177,12 @@ def parse(docstring):
   state.returns.lines = []
   state.yields.lines = []
   state.raises.lines = []
+  state.line1 = None
```

Comment:
Since you've written this with top-level state rather than per-arg state, I'm concerned about the possibility of state from the first arg leaking into the processing of the second arg.

Reply:
I do not see how it would leak. The only reason I did this with top-level state is otherwise we would have to repeat the below lines twice after line 405 and line 412. Do you think we should go with it being in the arg states?

Reply:
Here's what I meant by possible leakage:

If e.g. state.line3_first_word_length is set from the first arg, and then we start parsing data from the second arg, then state.line3_first_word_length will still be set with its old value.

Reply:
Yes, my inclination is toward arg state rather than top-level state, but I admit I haven't thought it through fully yet.

Reply:
Oooh I get it now! I did face this issue but I believe that our code should probably reset the values back to `None` if there is no line2 or line3. 

```
        else:
          state.line3_first_word_length = None
      else:
        state.line2_first_word_length = state.line2_length = None
```
above code is from line 432.

Reply:
The `state.max_line_length` would leak the max line length from arg1 to arg2 but I think we want that right?

We can move everything else except the max_line_length to the arg state. Wdyt?

Reply:
Makes sense to me.

We do have to be thoughtful then about whether we want to process the early args using the max_line_length available at the time that arg is ingested, or if we wait to the end to use the same max_line_length for all args.

Reply:
I chose to find the `state.max_line_length` right at the start (max line length of the whole docstring) and have updated the code to be per-arg state.